src/libcore/num/cast.rs

```diff
@@ -0,0 +1,86 @@
+use fmt;
+
+/// Internal trait for APIs like `i32::cast`.
+#[unstable(feature = "num_cast_internals", issue = "0")]
+pub trait Cast<T>: Sized {
+    #[doc(hidden)]
+    fn cast(t: T) -> Result<Self, CastError>;
+}
+
+/// Error type returned from APIs like `i32::cast`, indicates that a cast could
+/// not be performed losslessly.
+#[unstable(feature = "num_cast", issue = "0")]
+#[derive(Debug)]
+pub struct CastError(());
```

Comment:
It's pretty difficult to create a blanket policy that works everywhere, but this is intended for use in APIs that are already dealing with calls to libc and such and are almost guaranteed to already be returning `Result`.